tiledb/sm/query/ast/test/unit-query-ast.cc

```diff
@@ -0,0 +1,356 @@
+
+  auto combined_and4 = combined_and3->combine(
+      query_condition5, QueryConditionCombinationOp::AND);
+  CHECK(
+      ast_node_to_str(combined_and4) ==
+      "(foo NE 01 00 00 00 AND foo NE 03 00 00 00 AND foo NE 05 00 00 00 AND "
+      "foo NE 07 00 00 00 AND foo NE 09 00 00 00)");
+}
+
+TEST_CASE(
+    "ASTNode Constructors, adding simple clauses to an OR tree",
+    "[QueryCondition][ast][constructor]") {
+  // foo = 0 || foo = 2 || foo = 4 || foo = 6 || foo = 8
+  std::string field_name1 = "foo";
+  int val1 = 0;
+  auto query_condition1 = tdb_unique_ptr<ASTNode>(tdb_new(
+      ASTNodeVal,
+      std::string(field_name1),
+      &val1,
+      sizeof(int),
+      QueryConditionOp::EQ));
+  CHECK(ast_node_to_str(query_condition1) == "foo EQ 00 00 00 00");
+
+  std::string field_name2 = "foo";
+  int val2 = 2;
+  auto query_condition2 = tdb_unique_ptr<ASTNode>(tdb_new(
+      ASTNodeVal,
+      std::string(field_name2),
+      &val2,
+      sizeof(int),
+      QueryConditionOp::EQ));
+  CHECK(ast_node_to_str(query_condition2) == "foo EQ 02 00 00 00");
+
+  std::string field_name3 = "foo";
+  int val3 = 4;
+  auto query_condition3 = tdb_unique_ptr<ASTNode>(tdb_new(
+      ASTNodeVal,
+      std::string(field_name3),
+      &val3,
+      sizeof(int),
+      QueryConditionOp::EQ));
+  CHECK(ast_node_to_str(query_condition3) == "foo EQ 04 00 00 00");
+
+  std::string field_name4 = "foo";
+  int val4 = 6;
+  auto query_condition4 = tdb_unique_ptr<ASTNode>(tdb_new(
+      ASTNodeVal,
+      std::string(field_name4),
+      &val4,
+      sizeof(int),
+      QueryConditionOp::EQ));
+  CHECK(ast_node_to_str(query_condition4) == "foo EQ 06 00 00 00");
+
+  std::string field_name5 = "foo";
+  int val5 = 8;
+  auto query_condition5 = tdb_unique_ptr<ASTNode>(tdb_new(
+      ASTNodeVal,
+      std::string(field_name5),
+      &val5,
+      sizeof(int),
+      QueryConditionOp::EQ));
+  CHECK(ast_node_to_str(query_condition5) == "foo EQ 08 00 00 00");
+
+  auto combined_or1 = query_condition1->combine(
+      query_condition2, QueryConditionCombinationOp::OR);
+  CHECK(
+      ast_node_to_str(combined_or1) ==
+      "(foo EQ 00 00 00 00 OR foo EQ 02 00 00 00)");
+
+  auto combined_or2 =
+      combined_or1->combine(query_condition3, QueryConditionCombinationOp::OR);
+  CHECK(
+      ast_node_to_str(combined_or2) ==
+      "(foo EQ 00 00 00 00 OR foo EQ 02 00 00 00 OR foo EQ 04 00 00 00)");
+  auto combined_or3 =
+      combined_or2->combine(query_condition4, QueryConditionCombinationOp::OR);
+  CHECK(
+      ast_node_to_str(combined_or3) ==
+      "(foo EQ 00 00 00 00 OR foo EQ 02 00 00 00 OR foo EQ 04 00 00 00 OR foo "
+      "EQ 06 00 00 00)");
+  auto combined_or4 =
+      combined_or3->combine(query_condition5, QueryConditionCombinationOp::OR);
+  CHECK(
+      ast_node_to_str(combined_or4) ==
+      "(foo EQ 00 00 00 00 OR foo EQ 02 00 00 00 OR foo EQ 04 00 00 00 OR foo "
+      "EQ 06 00 00 00 OR foo EQ 08 00 00 00)");
```

Comment:
Just to add a little more coverage to the test -- could we use different operators in the different clauses instead of the same one for all the clauses? (Similarly in unit-QueryCondition).